pywebpack/__init__.py

```diff
@@ -7,7 +7,289 @@
+It is **important** to name assets with distinct filenames, or create a
+namespaced subfolder to contain your static files, so that name collisions
+will be avoided.
+
+In our main script, let's add two bundles (called ``mainsite`` and
+``backoffice``)::
+
+    from pywebpack.bundle import WebpackBundle
+    mainsite = WebpackBundle(
+        './modules/mainsite/static',
+        entry={
+            'mainsite-base': './js/mainsite-base.js',
+            'mainsite-products': './js/mainsite-products.js',
+        },
+        dependencies={
+            'jquery': '^3.2.1'
+        }
+    )
+
+    backoffice = WebpackBundle(
+        './modules/backoffice/static',
+        entry={
+            'backoffice-base': './js/backoffice-base.js',
+            'backoffice-admin': './js/backoffice-admin.js',
+        },
+        dependencies={
+            'jquery': '^3.1.0'
+        }
+    )
+
+A :class:`~pywebpack.bundle.WebpackBundle` requires the path to the static
+files of the module, an entry for each asset with its relative path and any
+extra npm package.
+
+Then, we create a project :class:`~pywebpack.project.WebpackBundleProject`
+for our bundles::
+
+    from pywebpack.project import WebpackBundleProject
+    project = WebpackBundleProject(
+        working_dir='build',  # where config and assets files will be copied
+        project_template_dir='buildconfig',  #`webpack.config.js` location
+        bundles=[mainsite, backoffice]
+    )
+
+Pywebpack will generate a ``config.json`` which will contain all the assets
+with their paths. Each bundle entry will be a `webpack entry`_.
+
+As last step, in our ``webpack.config.js``, we set the entry to the
+``config.json`` so webpack will build each asset::
+
+    var path = require('path');
+    var config = require('./config')
+
+    module.exports = {
+        context: path.resolve(__dirname),
+        entry: config.entry,
+        output: {
+            filename: '[name].js',
+            path: path.resolve(__dirname, 'dist')
+        }
+    };
+
+When executing ``project.buildall()``, Pywebpack will copy all the files
+contained in the ``project_template_dir`` folder and all the assets of each
+bundle to the ``working_dir`` folder. Then, it will install ``npm`` packages
+and run ``webpack`` to build the assets.
+
+Node dependencies resolution
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Bundles can declare extra npm packages needed to build the assets. Pywebpack
+supports ``dependencies``, ``devDependencies`` and ``peerDependencies``.
+
+Dependencies will be merged to a common list. In case the same dependency is
+defined multiple times with different version, only the highest version that
+satisfies all of them is kept. For more information, see the documentation of
+`node-semver`_.
+
+Extension points
+----------------
+
+With Pywebpack, we can "expose" bundles of our module so we can dynamically
+build assets of other installed modules.
+
+Let's define a module and expose it as an entry point. In
```

Comment:
entry point -> Python entry point ?

webpack entry vs python entry point can be a bit confusing :-)